components/org.wso2.carbon.identity.application.authenticator.biometric.servlet/src/main/java/org/wso2/carbon/identity/application/authenticator/biometric/servlet/servlet/BiometricServlet.java

```diff
@@ -0,0 +1,114 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.wso2.carbon.identity.application.authenticator.biometric.servlet.servlet;
+
+import com.google.gson.Gson;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.authenticator.biometric.servlet.javascript.flow.WaitStatusResponse;
+import org.wso2.carbon.identity.application.authenticator.biometric.servlet.model.WaitStatus;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.HashMap;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Component class for implementing the Biometric servlet.
+ */
+public class BiometricServlet extends HttpServlet {
+    private static final Log log = LogFactory.getLog(BiometricServlet.class);
+    private static final String DEVICE_TYPE = "deviceType";
+    private static final String MOBILE = "mobile";
+    private static final String WEB = "web";
+    private static final String SESSION_DATA_KEY_MOBILE = "sessionDataKeyMobile";
+    private static final String SESSION_DATA_KEY_WEB = "sessionDataKeyWeb";
+    private static final String CHALLENGE_MOBILE = "challengeMobile";
+    private HashMap<String, String> updateStatus = new HashMap<>();
+
+    @Override
+    protected void doGet(HttpServletRequest request,
+                         HttpServletResponse response) throws IOException {
+        doPost(request, response);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest request,
+                          HttpServletResponse response) throws IOException {
+
+        WaitStatusResponse waitResponse = null;
+        if (request.getParameterMap().containsKey(DEVICE_TYPE)) {
+            String deviceType = request.getParameter(DEVICE_TYPE);
+            waitResponse = new WaitStatusResponse();
+            if (deviceType.equals(MOBILE)) {
+                if ((request.getParameterMap().containsKey(SESSION_DATA_KEY_MOBILE)) &&
+                        request.getParameterMap().containsKey(CHALLENGE_MOBILE)) {
+                    String sessionDataKeyMobile = request.getParameter(SESSION_DATA_KEY_MOBILE);
+                    String challengeMobile = request.getParameter(CHALLENGE_MOBILE);
+                    log.info("challenge mobile parameter : " + challengeMobile);
+                    log.info("sdk mobile parameter : " + sessionDataKeyMobile);
+                    updateStatus.put(sessionDataKeyMobile, challengeMobile);
+                    log.info("table is: " + updateStatus);
+                    response.setContentType("text/html");
+                    response.setStatus(200);
+                    if (log.isDebugEnabled()) {
+                        log.debug("received the mobile SDK and challenge !");
```

Comment:
Since this is a debug log, let's add what are the received values here. Also, there is redundant whitespace at the end.